### ISSUE TITLE
automate update of certifi

### ISSUE THREAD

**Issue by benoitc** [certifi/erlang-certifi#30]
we should automate the update of certifi to make sure we are using the last updates from Mozilla.

cc @fenollp

**Comment by fenollp** [comment]
That's really just up to the TravisCI owner: https://docs.travis-ci.com/user/cron-jobs/
Once that job is set up & there's a diff the build will fail. Maybe there's an email notif but we should maybe look at something like IfThisThenThat to trigger the openning of an issue or directly of a PR.

I don't know a good solution but I'm very interested in something that's automatic! (while safe).

**Comment by benoitc** [comment]
i think it should be feasible to create a release directly sent to hex.pm one tagged. I will have a look

**Comment by benoitc** [comment]
looking at #32 i wonder if we could not have some kind of warning system included in certifi instead of rebuilding it:

* certifi is updated regularly using an update hooks
* a background daemon would check the last version in hex or something and log a warning if it needs to be updated. Then up to the user it would redeploy the app or dl a new file and do what #32 aims to provide.

Thoughts?

**Comment by fenollp** [comment]
I think a @daily build of master that cuts a minor release if there's a non-empty git diff (so no user interaction) would be best.
Not sure which kind of scrutiny you want here. I'd prefer fully automated/open/inpectable builds & deploys personally.

**Comment by paulo-ferraz-oliveira** [comment]
Two things to consider in the scope of this Issue:

a. automatic updates via `cron` aren't working since `.travis.yml` states `TRAVIS_OTP_RELEASE=20.2`, which is not declared in `otp_release` (in the same file)
b. it is possible that `erlang-certifi` stops using Travis CI soon

This Issue requires a **refresh** 😄

**Comment by paulo-ferraz-oliveira** [comment]
Regarding 2., which now seems closer to the complete objective (getting rid of Travis CI), how does this affect 1.? Do we still want `certifi` to build automatically? I'm assuming the goal is to:
1. check the "ca bundle" origin,
2. build, tag (git), publish (Hex) and release (GitHub)
Also, what does this mean for consumers of `master`. Do we assume that everybody's using a "lock" file and somehow _updating_ from `master` is considered innocuous?

**Comment by paulo-ferraz-oliveira** [comment]
@tsloughter, as of late Sep'21, https://github.com/g-andrade/tls_certificate_check/pulls?q=is%3Apr+is%3Aclosed is doing what I believe is what you're suggesting. Maybe `erlang-certifi` could do the same? ie. if I got your suggestion write, the automation would be in "importing the certificates", "creating a pull request", with tests (they won't run automatically, either) and merged by a human (no automatic releases).

**Comment by tsloughter** [comment]
Yea, looks like it.

**Comment by benoitc** [comment]
> Cool. But I would argue it isn't really an external service since it is the same service hosting the code and running the CI already :). And it is just a daily job to diff and open a PR if necessary, sounds like overkill to require an actual external service with full control for that.
> 
> Using this method is also what I found `castore` to be doing when I went looking for examples https://github.com/elixir-mint/castore/blob/master/.github/workflows/outdated.yml

Right. We can do that indeed. There are not that many updates (unless you're trying to do weird stuff in incorporating the certificates) so that should work.

By itself I think it should also be interresting to not rely totally on mkcert.org but that a second step. I will work on a change during the week unless someone take care of it first.

**Comment by benoitc** [comment]
@paulo-ferraz-oliveira well I already get myself an email when the bundle is changing so having a pr or not doesn't change anything. Plus useful certs bundled don't change that often. The point is more to create and send an updated version or trigger an external servuce to fetch it when needed.

**Comment by benoitc** [comment]
> @paulo-ferraz-oliveira well I already get myself an email when the bundle is changing so having a pr or not doesn't change anything. Plus useful certs bundled don't change that often. The point is more to create and send an updated version or trigger an external service to fetch it when needed.


> @tsloughter, as of late Sep'21, https://github.com/g-andrade/tls_certificate_check/pulls?q=is%3Apr+is%3Aclosed is doing what I believe is what you're suggesting. Maybe `erlang-certifi` could do the same? ie. if I got your suggestion write, the automation would be in "importing the certificates", "creating a pull request", with tests (they won't run automatically, either) and merged by a human (no automatic releases).

If an action can be triggerred by a webhook then maybe we can pass also some information to validate automatically that the bundle signature is OK during tests and then automatically trigger a release. Since I already have an automatic way to be notified when the bundle is updated that would be helpful.  I will look if we can do that .

**Comment by tofran** [comment]
@benoitc Regarding `mkcert.org`: why not just use the well trusted [CCADB](https://www.ccadb.org/resources) directly?

**Comment by neuromantik33** [comment]
We're currently encountering a show-stopper as letsencrypt root CA [expired](https://letsencrypt.org/docs/dst-root-ca-x3-expiration-september-2021/) as of 30-09-2021 and it is impossible to make HTTPS requests to sites signed with acme's generated letsencrypt certificates. An update of this library is very important!

**Comment by benoitc** [comment]
@neuromantik33 what version your library is using?

**Comment by benoitc** [comment]
> @benoitc Regarding `mkcert.org`: why not just use the well trusted [CCADB](https://www.ccadb.org/resources) directly?

certifi projects was based on mkcert.org. But we can revise it
